uni/lib/view/library/widgets/reservation_row.dart

```diff
@@ -0,0 +1,80 @@
+import 'package:flutter/material.dart';
+import 'package:intl/date_symbol_data_local.dart';
+import 'package:intl/intl.dart';
+import 'package:uni/model/entities/library_reservation.dart';
+import 'package:uni/model/entities/time_utilities.dart';
+
+class ReservationRow extends StatelessWidget {
+  final LibraryReservation? reservation;
```

Comment:
I think you can remove the nullable type, since this can only be built if a reservation exists in library_reservations_tab.dart. And it makes the code more readable too and with less null checks, in this file.